### ISSUE TITLE
fix-update MNX IDs according to MNX database latest release 4.0

### ISSUE THREAD

**Issue by feiranl** [SysBioChalmers/yeast-GEM#245]
### Description of the issue:
MetaNetX IDs in the model should be renewed based on the latest version 4.0 of MNX database, in this update, many MNX IDs are deprecated. Meanwhile MNX has updated the yeast model to yeast8.3.4 in their website, which reduced our work. We can update MNX IDs for both mets and rxns based on the info [yeast8.3.4]( https://www.metanetx.org/cgi-bin/mnxweb/mnet_info?model=yeastnet_v8_3_4).

Here is one example:
 MNX IDs are not longer existing: `MNXR125024` for `r_0465`, should be replaced by `MNXR153706`
 MNX met IDs are no longer existing: `MNXM162238` for `alpha-D-glucosamine 6-phosphate [cytoplasm]`, should be replaced by `MNXM730219`

 MNX IDs for `NH4+`, `water` , `O2` are also deprecated.

**I hereby confirm that I have:**
- [X] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---user) for running the model
- [X] Done this analysis in the `master` branch of the repository
- [X] Checked that a similar issue does not exist already
- [X] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about the issue

**Comment by feiranl** [comment]
@Hao-Chalmers @edkerk, What about also updating MNX database info stored in RAVEN toolbox, since in some function, we relies on 'mapIDsViaMNXref' to get IDs for other databases. Maybe user can chose which MNX version for mapping?

**Comment by Hao-Chalmers** [comment]
@feiranl good idea in providing different MNX versions in `mapIDsViaMNXref` function.

**Comment by edkerk** [comment]
These IDs should indeed be updated in yeast-GEM, and the relevant file should also be updated in RAVEN. What would be the benefit of supporting older MNX versions? I would not be recommended to use deprecated identifiers, and if a user would really want to do that, they can always pull an earlier commit / download an earlier release. So I suggest we only distribute and support the latest version.

**Comment by feiranl** [comment]
This update of MNX(4.0) updated so many IDs(including many basic chemicals such as water O2 and NH4+), if we only keep the latest 4.0 version, which may cause the function not working for every old models contain older version MNX IDs

**Comment by oxinabox** [comment]
If it is useful, once i finished working around  https://github.com/SysBioChalmers/yeast-GEM/issues/378
and a dozen or so metanetx inconsistencies

It's very easy for me to spit out a list of metabolite ids and matching metanetx 4.4 ids.
Since I have been building a process to resolve arbitrary GEMS against it, and testing on the yeast-GEM.

**Comment by edkerk** [comment]
Superseded: verify_annotations.py's \`drift\` check (added in #407) now automates exactly this -- detecting stored MetaNetX ids that have been deprecated in favour of a current one, for both reactions and metabolites. It currently finds 716, safe to batch-apply with \`--fix\`. See #408 for the current count and links to the full report.
